src/Countries/Countries.php

```diff
@@ -0,0 +1,1361 @@
+<?php
+
+/**
+ * Class that holds Countries list.
+ *
+ * @package EightshiftForms\Countries
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Countries;
+
+use EightshiftFormsVendor\EightshiftFormsUtils\Helpers\UtilsHooksHelper;
+
+/**
+ * Countries class.
+ */
+class Countries implements CountriesInterface
+{
+	/**
+	 * Get countries data set depending on the provided filter and default set.
+	 *
+	 * @param bool $useFullOutput Used to output limited output used for seetings and output.
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function getCountriesDataSet(bool $useFullOutput = true): array
+	{
+		$countries = $this->getCountriesList();
+
+		$output = [
+			'default' => [
+				'label' => \__('Default', 'eightshift-forms'),
+				'slug' => 'default',
+				'items' => $countries,
+				'codes' => \array_map(
+					static function ($item) {
+						return [
+							'label' => $item[0],
+							'value' => $item[1],
+						];
+					},
+					$countries
+				)
+			]
+		];
+
+		$alternative = [];
+		$filterName = UtilsHooksHelper::getFilterName(['block', 'country', 'alternativeDataSet']);
+		if (\has_filter($filterName)) {
+			$alternative = \apply_filters($filterName, []);
+		}
+
+		$alternativeOutput = [];
+
+		if ($alternative) {
+			foreach ($alternative as $value) {
+				$label = $value['label'] ?? '';
+				$slug = $value['slug'] ?? '';
+				$removed = isset($value['remove']) ? \array_flip($value['remove']) : [];
+				$onlyUse = isset($value['onlyUse']) ? \array_flip($value['onlyUse']) : [];
+				$changed = $value['change'] ?? [];
+
+				if (!$label || !$slug) {
+					continue;
+				}
+
+				$slug = \strtolower(\str_replace(' ', '-', $slug));
+
+				$alternativeOutput[$slug] = [
+					'label' => $label,
+					'slug' => $slug,
+					'items' => $countries,
+				];
+
+				$itemOutput = [];
+
+				foreach ($alternativeOutput[$slug]['items'] as $key => $item) {
+					$countryCode = $item[1] ? \strtolower($item[1]) : '';
+
+					// Only use.
+					if ($onlyUse && !isset($onlyUse[$countryCode])) {
+						continue;
+					}
+
+					// Remove item from list.
+					if (isset($removed[$countryCode])) {
+						continue;
+					}
+
+					// // Change label in the list.
+					foreach ($changed as $changedKey => $changedValue) {
+						if ($countryCode === $changedKey) {
+							$item[0] = $changedValue;
+						}
+					}
+
+					$itemOutput[] = $item;
+				}
+
+
+				$alternativeOutput[$slug]['items'] = $itemOutput;
+			}
+		}
```

Comment:
Ok, so I extracted this out to a separate file, then wrote a test for this by using 

```php
[
    [
        'label' => \__('China list'),
        'slug' => 'china-list',
        'change' => [
            'tw' => 'Taiwan',
        ],
    ],
]
```
As the alternative dataset, and the test worked (`Taiwan, China` was replaced with just `Taiwan`). The simplified code is:

```suggestion
		if ($alternative) {
			foreach ($alternative as $value) {
				$label = $value['label'] ?? '';
				$slug = $value['slug'] ?? '';
				if (!$label || !$slug) {
					continue;
				}

				$slug = \strtolower(\str_replace(' ', '-', $slug));
				$alternativeOutput[$slug] = [
					'label' => $label,
					'slug' => $slug,
					'items' => [],
				];

				$removed = $value['remove'] ?? [];
				$onlyUse = $value['onlyUse'] ?? [];
				$changed = $value['change'] ?? [];

				foreach ($countries as $item) {
					$countryCode = \strtolower($item[1]);

					if ((!empty($onlyUse) && !in_array($countryCode, $onlyUse)) ||
						in_array($countryCode, $removed)) {
						continue;
					}

					if (array_key_exists($countryCode, $changed)) {
						$item[0] = $changed[$countryCode];
					}

					$alternativeOutput[$slug]['items'][] = $item;
				}
			}
		}
```